### PR TITLE
Windows Installer (Silent mode)

### PR DESCRIPTION
**Pull request by thiagoftsm**
##### Summary
Fixes https://github.com/netdata/netdata/issues/18533

aa

The previous image was made before to attach option to accept the licenses.

##### Test Plan

1. Compile this branch
2. Generate package.
3. Install it using double-click on installer.
4. Open a terminal as administrator, and install it locally as described in the documentation brought in this PR.

##### Additional Information

<details> <summary>For users: How does this change affect me?</summary>
  
</details>

### CODE REVIEW

File: packaging/windows/WINDOWS_INSTALLER.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+- Welcome: The initial screen provides details about the actions the installer will perform.
+- UI License: This screen presents the Netdata Cloud UI. You must accept the license to proceed.
+- GPL 3 License: You will then be presented with the GPL 3 License, which governs netdata software.
+- Destination: By default, Netdata will be installed in `C:\Program Files\Netdata`. If you prefer a different directory, 
+  you can change it here.
+- Installation: At this stage, the software will be installed in the selected directory.
+- Claim: If you haven’t claimed your agent yet, this screen will allow you to
+  [claim](https://github.com/netdata/netdata/blob/master/src/claim/README.md) it. The following fields are available:
+ 
+    - Token: Your Netdata Cloud token.
```
Comment (Ancairon) [inline packaging/windows/WINDOWS_INSTALLER.md:31]:
```suggestion
    - Token: Your Netdata Cloud Space's claim token.
```

File: packaging/windows/WINDOWS_INSTALLER.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    - Proxy: (Optional) The proxy address used to communicate with Netdata Cloud.
+    - Insecure connection: By default, Netdata verifies the host certificate. If this option is selected,
+      certificate verification will be disabled.
+    - Open terminal: Selecting this option will open the `MSYS2 `terminal at the end of the installation.
+- Finish: The installation completes.  
+
+## Silent Mode
+
+For faster installation, the silent mode does not display the 
+[cloud license](https://raw.githubusercontent.com/netdata/netdata/master/src/web/gui/v2/LICENSE.md),
```
Comment (Ancairon) [inline packaging/windows/WINDOWS_INSTALLER.md:42]:
```suggestion
[Netdata Cloud UI License](src/web/gui/v2/LICENSE.md),
```

File: packaging/windows/WINDOWS_INSTALLER.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+or the [GPL3 license](https://www.gnu.org/licenses/gpl-3.0.txt). However, you must use the `/A` option to indicate that
```
Comment (Ancairon) [inline packaging/windows/WINDOWS_INSTALLER.md:43]:
```suggestion
nor the [GPL3 License](https://www.gnu.org/licenses/gpl-3.0.txt). However, you must use the `/A` option to indicate that
```

File: packaging/windows/WINDOWS_INSTALLER.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+you accept both licenses before proceeding.
```
Comment (Ancairon) [inline packaging/windows/WINDOWS_INSTALLER.md:44]:
```suggestion
you accept both Licenses before proceeding.
```

File: packaging/windows/WINDOWS_INSTALLER.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+In silent mode, the following options are available without additional arguments:
+
+- `/S`: Runs the installer in silent mode.
+- `/A`: Accepts all licenses required by Netdata. Without this option, the installer will not proceed.
```
Comment (Ancairon) [inline packaging/windows/WINDOWS_INSTALLER.md:49]:
```suggestion
- `/A`: Accepts all Licenses required by Netdata. Without this option, the installer will not proceed.
```

File: packaging/windows/WINDOWS_INSTALLER.md [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+- `/D`: Specifies the destination directory.
+- `/T`: Opens the `MSYS2` terminal after installation is complete.
+- `/I`: Accepts insecure connections, meaning the hostname will not be verified.
+
+Additionally, the following options require arguments:
+
+- `/TOKEN=`: Specifies your Netdata Cloud token.
```
Comment (Ancairon) [inline packaging/windows/WINDOWS_INSTALLER.md:56]:
```suggestion
- `/TOKEN=`: Specifies your Netdata Cloud Space's claim token.
```

File: packaging/windows/WINDOWS_INSTALLER.md [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+- `/ROOMS=`: Specifies the rooms to which you want to connect the host. Multiple rooms can be specified by separating
```
Comment (Ancairon) [inline packaging/windows/WINDOWS_INSTALLER.md:57]:
```suggestion
- `/ROOMS=`: Specifies the Room IDs to which you want to connect the node to. Multiple rooms can be specified by separating
```

**Review by Ancairon — COMMENTED** [review]
Grammar and syntax changes, also terminology fixes on Docs

**Comment by thiagoftsm** [comment]
> Grammar and syntax changes, also terminology fixes on Docs

Thank you, I merged all.

**Comment by stelfrag** [comment]
The silent installation works `netdata-installer.exe /s /a` however there is no immediate feedback on the command prompt

Just noting the current behavior -- the output is 
```
C:\>\msys64\home\stelios\netdata\packaging\windows\netdata-installer.exe /S /A
C:\>
C:\>
C:\>
C:\>echo %ERRORLEVEL%
0
C:\>Netdata installed with success.\n
```
Note that the installer returns to the command prompt immediately (the return code is 0) but installation is not finished

There is an `Netdata installed with success.\n` message displayed when it is finished

You can also also start multiple installations -- we need to somehow check this

**Comment by thiagoftsm** [comment]
> The silent installation works `netdata-installer.exe /s /a` however there is no immediate feedback on the command prompt
> 
> Just noting the current behavior -- the output is

> ```
> C:\>\msys64\home\stelios\netdata\packaging\windows\netdata-installer.exe /S /A
> C:\>
> C:\>
> C:\>
> C:\>echo %ERRORLEVEL%
> 0
> C:\>Netdata installed with success.\n
> ```
> 
The behavior is happening because `netdata-installer.exe` is a graphic software. I am going to modify the installer to use mutex avoiding two instance running in parallel.

**Comment by stelfrag** [comment]
A few comments

- Installer can now detect multiple instances (although delayed async message on the next prompt)
- `netdata-installer /s` will request (again via async message) that 

```
c:\msys64\home\user\netdata\packaging\windows>netdata-installer.exe /s

c:\msys64\home\user\netdata\packaging\windows>
c:\msys64\home\user\netdata\packaging\windows>
c:\msys64\home\user\netdata\packaging\windows>You must accept the licenses (/A) to continue.
```

- `netdata-installer.exe /a` will bring up the UI installer (but the /a is not respected so UI will prompt to accept) 
   - Not sure if we need to change this behavior

**Comment by thiagoftsm** [comment]
>     * Installer can now detect multiple instances (although delayed async message on the next prompt)
As I had explained, unfortunately we are starting a Window and we only print messages after the software exist and we return to terminal.

>     * `netdata-installer /s` will request (again via async message) that
The same of previous.
>     * `netdata-installer.exe /a` will bring up the UI installer (but the /a is not respected so UI will prompt to accept)

Considering that person is not using silent mode, I understand they should see the licenses, in reality I think we should add a checkbox for them. :thinking:

**Pull request merged**